examples/BackupRestoreTags/tagutil.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/TheJumpCloud/jcapi"
+)
+
+const (
+	URL_BASE string = "https://console.jumpcloud.com/api"
+)
+
+func fileExists(fileName string) (exists bool) {
+	if _, err := os.Stat(fileName); err == nil {
+		exists = true
+	}
+
+	return
+}
+
+func readTagFile(backupFileName string) (tagData jcapi.JCTag, err error) {
+	fileData, err := ioutil.ReadFile(backupFileName)
+	if err != nil {
+		return tagData, fmt.Errorf("Could not read file '%s', err='%s'", backupFileName, err)
+	}
+
+	err = json.Unmarshal(fileData, &tagData)
+	if err != nil {
+		return tagData, fmt.Errorf("Could not unmarshal tag data from file '%s', err='%s'", backupFileName, err)
+	}
+
+	return
+}
+
+func writeTagFile(tagFileName string, tag jcapi.JCTag) (err error) {
+	tagData, err := json.Marshal(&tag)
+	if err != nil {
+		return fmt.Errorf("Cannot marshal tag data for tag '%s', err='%s'", tag.Name, err)
+	}
+
+	if fileExists(tagFileName) {
+		return fmt.Errorf("Tag backup file '%s' exists in the current working directory, please move it before backing up.", tagFileName)
+	}
+
+	err = ioutil.WriteFile(tagFileName, tagData, 0600)
+	if err != nil {
+		return fmt.Errorf("Could not write tag data for tag '%s' to file '%s', err='%s'", tag.Name, tagFileName, err)
+	}
+
+	return
+}
+
+func getBackupDirName() (backupDirName string) {
+	t := time.Now()
+	tz, _ := t.Zone()
+
+	backupDirName = "tagBackup-" + fmt.Sprintf("%d-%02d-%02dT%02d:%02d:%02d_%s", t.Year(), t.Month(), t.Day(),
+		t.Hour(), t.Minute(), t.Second(), tz)
+
+	return
+}
+
+func backupTags(tags []jcapi.JCTag, tagRegex string) (backedUpCount, tagCount int, backupDirName string, err error) {
+	regex, err := regexp.Compile(tagRegex)
+	if err != nil {
+		return 0, 0, "", fmt.Errorf("Could not compile regex, err='%s'", err)
+	}
+
+	backupDirName = getBackupDirName()
+
+	err = os.MkdirAll(backupDirName, 0700)
+	if err != nil {
+		return 0, 0, "", fmt.Errorf("Could not create backup directory '%s', err='%s'", backupDirName, err)
+	}
+
+	for _, value := range tags {
+		if regex.FindString(value.Name) != "" {
+			err := writeTagFile(filepath.Join(backupDirName, value.Name+".jcback"), value)
+			if err != nil {
+				return 0, 0, "", fmt.Errorf("Could not write tag '%s', err='%s'", value.Name, err)
+			}
+
+			backedUpCount++
+		}
+	}
+
+	tagCount = len(tags)
+
+	return
+}
+
+func findTagByName(tags []jcapi.JCTag, tagName string) (idx int) {
+	for idx, value := range tags {
+		if value.Name == tagName {
+			return idx
+		}
+	}
+
+	return -1
```

Comment:
I'm fine with that.  I was thinking of how `database/sql` returns an error for no rows.